packages/mui/src/date-time-pickers/wrapper.tsx

```diff
@@ -20,22 +26,57 @@ type PickerProps<
   Out = In,
 > = P extends typeof TimePicker
   ? TimePickerProps<In, Out>
-  : P extends typeof DatePicker
-  ? DatePickerProps<In, Out>
-  : P extends typeof DateTimePicker
-  ? DateTimePickerProps<In, Out>
-  : never;
+  : (P extends typeof DatePicker
+      ? DatePickerProps<In, Out>
+      : P extends typeof DateTimePicker
+      ? DateTimePickerProps<In, Out>
+      : never) & { multiLocale?: boolean };
 
 type Props<P extends PickerTypes, In, Out> = SubsetPartial<
   PickerProps<P, In, Out>,
   'renderInput'
 >;
+
 export { Props as WrappedPickerProps };
 
 export function WrapPicker<P extends PickerTypes>(Picker: P) {
   function WrappedPicker<In, Out>(props: Props<P, In, Out>) {
+    const [locale, setLocale] = useState<Locale>(Locale.defaultLocale);
+    const handleLocaleChange = () => {
+      setLocale((prevLocale) =>
+        prevLocale === Locale.en ? Locale.fa : Locale.en,
+      );
+    };
+    const localeText = useLocaleText();
+
     const commonPickerProps = usePickerProps<In, Out>();
 
+    // @ts-ignore
+    if ('multiLocale' in props && props.multiLocale) {
+      let changedProps = mergeDeepRight(props, {
+        components: {
+          ActionBar: MultiLocalePickersActionBar,
+        },
+        componentsProps: {
+          actionBar: { locale, onLocaleChange: handleLocaleChange },
+        },
+      }) as unknown as PickerProps<P, In, Out>;
+
+      return (
```

Comment:
always return this tree with changedProps or without props

Reply:
I will add MultiLocaleWrapper in next commit.

Reply:
MultiLocaleWrapper is deleted right now!